docs/en/sql-reference/functions/splitting-merging-functions.md

```diff
@@ -33,6 +33,13 @@ Returns an array of selected substrings. Empty substrings may be selected when:
 
 Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).
 
+:::note
+The behavior of parameter `max_substrings` changed starting with ClickHouse v22.11. In versions older than that, `max_substrings` > 0 meant that `max_substring`-many splits were performed and that the remainder of the string was returned as the final element of the list.
+For example,
+- in v22.10: `SELECT splitByChar('=', 'a=b=c=d', 2); -- ['a','b','c=d']`
```

Comment:
why did we change it? the old behaviour is pretty common across other implementations.

Reply:
It was changed in PR #42673, see the discussion after https://github.com/ClickHouse/ClickHouse/pull/42673#issuecomment-1701190114